spec/20-http_request_header_format.md

```diff
@@ -282,7 +282,7 @@ A `key` MUST begin with a lowercase letter or a digit and contain up to 256 char
 
 ##### Value
 
-The value is an opaque string containing up to 256 printable ASCII [[!RFC0020]] characters (i.e., the range 0x20 to 0x7E) except comma (`,`) and (`=`). Note that this also excludes tabs, newlines, carriage returns, etc.
+The value is an opaque string containing up to 256 printable ASCII [RFC0020] characters (i.e., the range 0x20 to 0x7E) except comma (,) and (=) which must end with a character which is not a space (0x20). Note that this also excludes tabs, newlines, carriage returns, etc. This effectively means that all leading spaces MUST be preserved as part of the value and all trailing spaces MUST not be preserved.
```

Comment:
It is still allowed to preserve the OWS characters.

```suggestion
The value is an opaque string containing up to 256 printable ASCII [RFC0020] characters (i.e., the range 0x20 to 0x7E) except comma (,) and (=) which must end with a character which is not a space (0x20). Note that this also excludes tabs, newlines, carriage returns, etc. This effectively means that all leading spaces MUST be preserved as part of the value and all trailing spaces MAY not be preserved.
```

Reply:
Really? The grammar ends with nblk-chr. Doesn't that mean it's not ok to preserve trailing space?

Reply:
There is OWS allowed between kv pairs

```
key1=value1 , key2=value2
           ^this is optional whitespace, not part of the value, and may be preserved or dropped
```

Reply:
Ah maybe we're not on the same page on this

> not part of the value, and may be preserved or dropped

What did it mean to be not part of the value, but preserved? Does it mean that the value MUST be presented to user space (i.e., SDK) without trailing space but when propagating it's ok to keep or drop it? If so it makes sense and I'll add the caveat about propagation.

Reply:
Maybe this
```suggestion
The value is an opaque string containing up to 256 printable ASCII [RFC0020] characters (i.e., the range 0x20 to 0x7E) except comma (,) and (=) which must end with a character which is not a space (0x20). Note that this also excludes tabs, newlines, carriage returns, etc. This means that all leading spaces MUST be preserved as part of the value and all trailing spaces are considered to be optional whitespace characters not part of the value.
```

Reply:
Thanks this makes sense but is still a bit unclear about what it means, in practice, to be "considered to be optional whitespace". I added a point about propagation which is my understanding but I'm not sure. It's why I want to make sure this spec is clear :)